magicctapipe/scripts/lst1_magic/lst_m1_m2_cloud_correction.py

```diff
@@ -0,0 +1,349 @@
+            & (dl1_images["obs_id"] == obs_id)
+        )[0]
+
+        if len(inds_img) > 0:
+            for index_img in inds_img:
+                image = dl1_images["image"][index_img]
+                cleanmask = dl1_images["image_mask"][index_img]
+                peak_time = dl1_images["peak_time"][index_img]
+                image /= trans_pixels
+                corr_image = image.copy()
+                corr_image[~cleanmask] = 0
+
+                hillas_params = hillas_parameters(camgeom[telid], corr_image)
+                timing_params = timing_parameters(
+                    camgeom[telid], corr_image, peak_time, hillas_params
+                )
+                leakage_params = leakage_parameters(
+                    camgeom[telid], corr_image, signal_pixels
+                )
+                conc_params = concentration_parameters(
+                    camgeom[telid], corr_image, hillas_params
+                )
+
+                event_params = {
+                    **hillas_params,
+                    **timing_params,
+                    **leakage_params,
+                    **conc_params,
+                }
+
+                # Add real event information (assuming it's also a dictionary-like object)
+                event_info_dict = {
+                    "obs_id": obs_id,
+                    "event_id": event_id,
+                    "pointing_alt": pointing_alt,
+                    "pointing_az": pointing_az,
+                    "time_diff": time_diff,
+                    "n_pixels": n_pixels,
+                    "n_islands": n_islands,
+                }
+                event_params.update(event_info_dict)
+
+                all_params_list.append(event_params)
+
+        else:
+            raise ValueError("Error: 'inds_img' list is empty!")
+
+    df = pd.DataFrame(all_params_list)
+    return df
+
+
+def main():
+    """Main function."""
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--input_file",
+        "-i",
+        dest="input_file",
+        type=str,
+        required=True,
+        help="Path to an input .h5 DL1 data file",
+    )
+
+    parser.add_argument(
+        "--output_file",
+        "-o",
+        dest="output_file",
+        type=str,
+        default="./data",
+        help="Path to a directory where to save an output corrected DL1 file",
+    )
+
+    args = parser.parse_args()
+
+    subarray_info = SubarrayDescription.from_hdf(args.input_file)
+
+    tel_descriptions = subarray_info.tel
+    camgeom = {}
+    for telid, telescope in tel_descriptions.items():
+        camgeom[telid] = telescope.camera.geometry
+
+    optics_table = read_table(
+        args.input_file, "/configuration/instrument/telescope/optics"
+    )
+    focal_eff = {}
+
+    for telid, telescope in tel_descriptions.items():
+        optics_row = optics_table[optics_table["optics_name"] == telescope.name]
+        if len(optics_row) > 0:
```

Comment:
you really need this focal length for the code to work, so if optics_row is empty you should raise an exception